ex1/dates_times.py

```diff
@@ -32,13 +32,47 @@
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
+import calendar as cal
+
+import datetime as date
+def last_of_month(dt):
+    return date.date(dt.year, dt.month, cal.monthrange(dt.year, dt.month)[1])
+    # -> as calender.monthrange(datetime) would return a list
+    # of the first weekday and the number of days of the submitted month,
+    # we just need the the second element of (0,31) ect.
+#print last_of_month(date.date.today())
+print last_of_month(date.date(2007, 5, 31))
```

Comment:
Its best not to have print keywords here since they will be executed when the module is imported.